app/screens/reaction_list/reaction_list.js

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {View} from 'react-native';
+import {intlShape} from 'react-intl';
+import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {
+    generateUserProfilesById,
+    getMissingUserIds,
+    getReactionsByName,
+    getSortedReactionsForHeader,
+    getUniqueUserIds,
+    sortReactions,
+} from 'app/utils/reaction';
+
+import ReactionHeader from './reaction_header';
+import ReactionRow from './reaction_row';
+
+import {ALL_EMOJIS} from 'app/constants/emoji';
+
+export default class ReactionList extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            getMissingProfilesByIds: PropTypes.func.isRequired,
+        }).isRequired,
+        navigator: PropTypes.object,
+        reactions: PropTypes.array.isRequired,
+        theme: PropTypes.object.isRequired,
+        teammateNameDisplay: PropTypes.string,
+        userProfiles: PropTypes.array,
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+        const {reactions, userProfiles} = props;
+
+        const reactionsByName = getReactionsByName(reactions);
+
+        this.state = {
+            allUserIds: getUniqueUserIds(reactions),
+            reactions,
+            reactionsByName,
+            selected: ALL_EMOJIS,
+            sortedReactions: sortReactions(reactionsByName),
+            sortedReactionsForHeader: getSortedReactionsForHeader(reactionsByName),
+            userProfiles,
+            userProfilesById: generateUserProfilesById(userProfiles),
+        };
+
+        props.navigator.setOnNavigatorEvent(this.onNavigatorEvent);
+    }
+
+    static getDerivedStateFromProps(nextProps, prevState) {
+        let hasNewState = false;
+        const newState = {};
+        if (!nextProps.reactions.length !== prevState.reactions.length) {
+            const {reactions} = nextProps;
+            const reactionsByName = getReactionsByName(reactions);
+
+            newState.allUserIds = getUniqueUserIds(reactions);
+            newState.reactions = reactions;
+            newState.reactionsByName = reactionsByName;
+            newState.sortedReactions = sortReactions(reactionsByName);
+            newState.sortedReactionsForHeader = getSortedReactionsForHeader(reactionsByName);
+
+            hasNewState = true;
+        }
+
+        if (nextProps.userProfiles !== prevState.userProfiles) {
+            newState.userProfilesById = generateUserProfilesById(nextProps.userProfiles);
+
+            hasNewState = true;
+        }
+
+        return hasNewState ? newState : null;
```

Comment:
I think you can just return `newState` here regardless of whether or not there's anything changed in it, can't you? Tracking `hasNewState` doesn't seem necessary

Reply:
Okay will try to change that.  I'm hesitant to do it at first as the docs says it should return an object to update the state, or null to update nothing.